### ISSUE TITLE
NordVPN Api

### ISSUE THREAD

**Issue by lbr88** [gjedeer/nordvpn#1]
Not really and issue/bug but just wanted to let you know that you could fetch the servers from the API here in JSON format:
https://nordvpn.com/api/server

**Comment by gjedeer** [comment]
Wooooah dude. How did you find that?

**Comment by lbr88** [comment]
Trial and error basically. I was making a shell script and systemd service file to easily download and update ovpn configs. Unfortunately i discovered it after i made my script.
I just found this also:
https://bitbucket.org/nordial/nordapi/src/dd4e918e84b66a7842cbbf2262ad4019ead2e877/nordtoy.py?at=master&fileviewer=file-view-default
don't know how much of it works or not.
Edit: Doesn't look like any of those works.

**Comment by gjedeer** [comment]
Good research. I haven't found any of these and I did all I could to avoid parsing HTML.

**Comment by bguidolim** [comment]
Maybe it's better for you case: https://nordvpn.com/api/server/stats

**Comment by gjedeer** [comment]
Yeah, maybe. I'm not maintaining this so you're welcome to fork and use the correct API.

**Comment by mrzool** [comment]
How on earth do people find these undocumented API endpoints?

Now that's seemingly the only way to get access to that vital information, since those asshats at NordVPN decided to [replace the HTML server list](https://mobile.twitter.com/NordVPN/status/935461778858921984) with [useless marketing copy](https://nordvpn.com/servers/germany/).

**Comment by lbr88** [comment]
Trial and error, or maybe sniffing the nordvpn windows client which uses the api heavily.

**Comment by mrzool** [comment]
@lbr88 well, you and @bguidolim solved a pretty pressing problem of mine by finding it and taking the time share it, so thanks a lot for that.

**Comment by dym-ok** [comment]
An easier way is looking into the iOS/Android app with the help of https://mitmproxy.org/

**Comment by mrzool** [comment]
Just wanted to mention briefly that thanks to those endpoints @lbr88 and @bguidolim posted  I was able to write a little script which works beautifully for finding the fastest NordVPN servers: https://github.com/mrzool/nordvpn-server-find

**Comment by gjedeer** [comment]
@mrzool https://github.com/gjedeer/nordvpn/commit/d40679a2b49b0e3ec4b08cf5af28b7225684b05f

**Comment by RevertIT** [comment]
Hmm seems like the list of IPs is not updated from https://nordvpn.com/api/server
